src/common/snippets/src/utils/tokenization_utils.cpp

```diff
@@ -0,0 +1,434 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <cassert>
+#include <climits>
+
+#include "snippets/utils/tokenization_utils.hpp"
+#include "snippets/remarks.hpp"
+
+namespace ov {
+namespace snippets {
+namespace utils {
+
+using namespace ov::snippets::op;
+using namespace ov::snippets::pass;
+
+namespace {
+auto has_result_child(const std::shared_ptr<const Node> &node) -> bool {
+    for (const auto& child : node->get_users()) {
+        if (ov::is_type<ov::opset1::Result>(child)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+auto get_num_result_children(const std::shared_ptr<const Node> &node) -> size_t {
+    size_t result = 0;
+    for (const auto& child : node->get_users()) {
+        if (ov::is_type<ov::opset1::Result>(child)) {
+            result++;
+        }
+    }
+    return result;
+}
+
+auto outputs_are_not_broadcastable(const std::shared_ptr<const Node>& node) -> bool {
+    const auto& outputs = node->outputs();
+    if (outputs.size() <= 1)
+        return false;
+    ov::PartialShape ref_shape = outputs.front().get_partial_shape();
+    bool success = true;
+    for (size_t i = 1; i < outputs.size() && success; i++) {
+        success &= ov::PartialShape::broadcast_merge_into(ref_shape, outputs[i].get_partial_shape(), ov::op::AutoBroadcastType::NUMPY);
+    }
+    return !success;
+}
+}  // namespace
+
+bool tokenize_node(const std::shared_ptr<ov::Node>& node, const SnippetsTokenization::Config& config) {
+    const auto getFusedNames = [](const std::shared_ptr<Node>& n) -> std::string {
+        auto rt_info = n->get_rt_info();
+        auto it = rt_info.find("originalLayersNames");
+        if (it != rt_info.end()) {
+            return it->second.as<std::string>() + ",";
+        }
+        return "";
+    };
+
+    auto create_single_node_subgraph = [&](const std::shared_ptr<Node> &node) {
+        auto subgraph = op::Subgraph::wrap_node_as_subgraph(node);
+        subgraph->get_rt_info()["originalLayersNames"] = getFusedNames(node) + node->get_friendly_name();
+        ov::replace_node(node, subgraph);
+        op::update_out_tensor_name(subgraph);
+    };
+
+    auto abort_with_strategy = [&](const std::string& message_reset, const std::string& message_abort = "") {
```

Comment:
```suggestion
    auto abort = [&]() {
```

Reply:
Good proposal, thanks. But I left a `const std::string& message` parameter: it is dumped using `remark` mechanism now.